samples/junit-tests/src/test/java/com/yelp/codegen/generatecodesamples/PostFileEndpointTest.kt

```diff
@@ -0,0 +1,179 @@
+package com.yelp.codegen.generatecodesamples
+
+import com.yelp.codegen.generatecodesamples.apis.FileApi
+import com.yelp.codegen.generatecodesamples.tools.MockServerApiRule
+import okhttp3.MediaType
+import okhttp3.RequestBody
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.RecordedRequest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+
+private data class FileApiStats(
+    var boundary: String? = null,
+    var errors: List<String>? = null,
+    var parts: List<MultiPartInfo>? = null
+)
+private data class MultiPartInfo(
+    val content: String? = null,
+    val contentDisposition: String? = null,
+    val contentLength: String? = null,
+    val contentType: String? = null
+)
+
+private fun processPart(part: String): MultiPartInfo {
+
+    val partLines = part.split("\r\n").let {
+        // part would look like "\r\nHeader-Name: Header-Value\r\n...\r\n\r\nbody\r\n"
+        // Here we're removing the first and last part as we know that are empty and
+        it.subList(1, it.size - 1)
+    }
+
+    val partLinesIterator = partLines.iterator()
+    val headers = mutableMapOf<String, String>()
+    for (headerLine in partLinesIterator) {
+        if (headerLine.isEmpty()) {
+            break
+        } else {
+            val (name, value) = headerLine.split(": ", limit = 2)
+            headers[name] = value
+        }
+    }
+
+    val content = partLinesIterator.asSequence().joinToString("\r\n")
```

Comment:
```suggestion
val content = part.split("\r\n")
            .drop(1)
            .dropLast(1)
            .joinToString("\r\n")
```

Reply:
See previous comment.

`content` should contain the content of the uploaded file, so we need to make sure that we're continuing iterating on `part` from the correct position

Dropping this as the discussion on the point above will provide valuable input for this as well (modifying headers extraction implies modifying body extraction)